app/javascript/mastodon/actions/notification_groups.ts

```diff
@@ -107,8 +117,15 @@ export const fetchNotificationsGap = createDataLoadingThunk(
 export const pollRecentNotifications = createDataLoadingThunk(
   'notificationGroups/pollRecentNotifications',
   async (_params, { getState }) => {
+    const activeFilter =
+      selectSettingsNotificationsQuickFilterActive(getState());
+
     return apiFetchNotifications({
       max_id: undefined,
+      exclude_types:
+        activeFilter === 'all'
+          ? selectSettingsNotificationsExcludedTypes(getState())
+          : excludeAllTypesExcept(activeFilter),
```

Comment:
Looks like a little duplication with `fetchNotificationsGap` above. How about adding a small utility function to compute the `exclude_types` parameter?